packages/webapp/src/components/Task/MovementTask/index.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useTranslation } from 'react-i18next';
+import { Controller, UseFormReturn } from 'react-hook-form';
+import ReactSelect from '../../Form/ReactSelect';
+import Input, { getInputErrors } from '../../Form/Input';
+import { hookFormMaxCharsValidation } from '../../Form/hookformValidationUtils';
+import styles from './styles.module.scss';
+import { useGetAnimalMovementPurposesQuery } from '../../../store/api/apiSlice';
+
+type PureMovementTaskProps = UseFormReturn;
+
+const PureMovementTask = (props: PureMovementTaskProps) => {
+  const {
+    control,
+    register,
+    watch,
+    formState: { errors },
+  } = props;
+
+  const { t } = useTranslation(['translation', 'message', 'animal']);
+  const { data: purposes = [] } = useGetAnimalMovementPurposesQuery();
+
+  const PURPOSE = `movement_task.purpose`;
+  const OTHER_PURPOSE_EXPLANATION = `movement_task.other_purpose_explanation`;
+
+  const selectedPurpose = watch(PURPOSE);
+
+  return (
+    <div className={styles.movementTaskDetailsContainer}>
+      <Controller
+        control={control}
+        name={PURPOSE}
+        rules={{ required: true }}
+        render={({ field: { onChange, value } }) => (
+          <ReactSelect
+            value={value}
+            label={t('ADD_TASK.MOVEMENT_VIEW.MOVEMENT_PURPOSE_LABEL')}
+            options={purposes.map((purpose) => ({
+              value: purpose.id,
+              label: t(`animal:PURPOSE.${purpose.key}`),
+            }))}
+            onChange={onChange}
+            placeholder={t('ADD_TASK.MOVEMENT_VIEW.MOVEMENT_PURPOSE_PLACEHOLDER')}
+            isMulti
+          />
+        )}
+      />
+      {selectedPurpose?.key === 'OTHER' && (
```

Comment:
`selectedPurpose` is an array, and it only has `value` and `label` 😉

Reply:
Ah thank you so much! I changed the code on my last commit and didn't re-test this bit, my bad. Will update!